modules/gcp/compute.go

```diff
@@ -7,146 +7,259 @@ import (
 	if err != nil {
-		return "", fmt.Errorf("Instances.Get(%s) got error: %v", instanceID, err)
+		t.Fatal(err)
 	}
+	return ip
+}
 
+// GetPublicIpE gets the public IP address of the given Compute Instance.
+func (c *Instance) GetPublicIpE(t *testing.T) (string, error) {
 	// If there are no accessConfigs specified, then this instance will have no external internet access:
 	// https://cloud.google.com/compute/docs/reference/rest/v1/instances.
-	if len(instance.NetworkInterfaces[0].AccessConfigs) == 0 {
-		return "", fmt.Errorf("Attempted to get public IP of Compute Instance %s, but that Compute Instance does not have a public IP address", instanceID)
+	if len(c.NetworkInterfaces[0].AccessConfigs) == 0 {
+		return "", fmt.Errorf("Attempted to get public IP of Compute Instance %s, but that Compute Instance does not have a public IP address", c.Name)
 	}
 
-	ip := instance.NetworkInterfaces[0].AccessConfigs[0].NatIP
+	ip := c.NetworkInterfaces[0].AccessConfigs[0].NatIP
 
 	return ip, nil
 }
 
-// GetLabelsForComputeInstance returns all the tags for the given Compute Instance.
-func GetLabelsForComputeInstance(t *testing.T, projectID string, zone string, instanceID string) map[string]string {
-	labels, err := GetLabelsForComputeInstanceE(t, projectID, zone, instanceID)
+// GetLabels returns all the tags for the given Compute Instance.
+func (c *Instance) GetLabels(t *testing.T) map[string]string {
+	return c.Labels
+}
+
+// GetZone returns the Zone in which the Compute Instance is located.
+func (c *Instance) GetZone(t *testing.T) string {
+	return ZoneUrlToZone(c.Zone)
+}
+
+// SetLabels adds the tags to the given Compute Instance.
+func (c *Instance) SetLabels(t *testing.T, labels map[string]string) {
+	err := c.SetLabelsE(t, labels)
 	if err != nil {
 		t.Fatal(err)
 	}
-	return labels
 }
 
-// GetLabelsForComputeInstanceE returns all the tags for the given Compute Instance.
-func GetLabelsForComputeInstanceE(t *testing.T, projectID string, zone string, instanceID string) (map[string]string, error) {
-	logger.Logf(t, "Getting Labels for Compute Instance %s", instanceID)
+// SetLabelsE adds the tags to the given Compute Instance.
+func (c *Instance) SetLabelsE(t *testing.T, labels map[string]string) error {
+	logger.Logf(t, "Adding labels to instance %s in zone %s", c.Name, c.Zone)
 
 	ctx := context.Background()
-
 	service, err := NewComputeServiceE(t)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	instance, err := service.Instances.Get(projectID, zone, instanceID).Context(ctx).Do()
-	if err != nil {
-		return nil, fmt.Errorf("Instances.Get(%s) got error: %v", instanceID, err)
+	req := compute.InstancesSetLabelsRequest{Labels: labels, LabelFingerprint: c.LabelFingerprint}
+	if _, err := service.Instances.SetLabels(c.projectID, c.GetZone(t), c.Name, &req).Context(ctx).Do(); err != nil {
+		return fmt.Errorf("Instances.SetLabels(%s) got error: %v", c.Name, err)
 	}
 
-	return instance.Labels, nil
+	return err
```

Comment:
`return nil`?

Reply:
Good catch. I had to fix this in other func as well.